lib/internal/modules/esm/hooks.js

```diff
@@ -174,7 +191,46 @@ class Hooks {
       const next = this.#chains.load[this.#chains.load.length - 1];
       ArrayPrototypePush(this.#chains.load, { __proto__: null, fn: load, url, next });
     }
-    return initialize?.(data);
+
+    ArrayPrototypePush(this.#loaderInstances, instance);
```

Comment:
```suggestion
    this.#customLoaders.set(registrationId, {
      __proto__: null,
      id: registrationId,
      globalPreload,
      initialize,
      resolve,
      load,
    });
```

I think the `__proto__: null` here is actually not necessary, but why not just in case.